File: core/src/main/java/org/elkd/core/cluster/ClusterUtils.java
package org.elkd.core.cluster;

import org.elkd.core.config.Config;
import org.elkd.shared.schemes.URI;

public class ClusterUtils {
  private ClusterUtils() { }

  public static Node buildSelfNode(final Config config) {
    final URI uri = URI.parseURIString(config.get(Config.KEY_HOST) + ":" + config.getAsInteger(Config.KEY_PORT));
    return new Node(uri);
  }
}

File: core/src/main/java/org/kerala/core/server/cluster/ClusterUtils.kt
package org.kerala.core.server.cluster

import org.kerala.core.Environment
import org.kerala.core.config.Config
import org.kerala.shared.schemes.URI

object ClusterUtils {
  fun buildSelfNode(): Node {
    val host: String = Environment.config[Config.KEY_HOST]
    val port: Int = Environment.config[Config.KEY_PORT]
    return Node(URI.parseURIString("$host:$port"))
  }
}
